### PR TITLE
Addresses

### PR DESCRIPTION
**Pull request by wanderer**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -135,31 +109,28 @@ module.exports = class Hypervisor {
...
-
     return this.tree.flush()
   }
 
   /**
    * regirsters a container with the hypervisor
    * @param {Class} Constructor - a Class for instantiating the container
    * @param {*} args - any args that the contructor takes
-   * @param {interger} typeId - the container's type identification ID
+   * @param {Interger} typeId - the container's type identification ID
```
Comment (sdtsui) [inline index.js]: 'inte~r~ger'

File: capsManager.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+   * fetching and waiting on ports
+   * @param {Object} caps
+   */
+  constructor (caps) {
+    this._storedCaps = caps
+    this.clist = new Set()
+  }
+
+  /**
+   * Stores a capability persistantly
```
Comment (sdtsui) [inline capsManager.js]: 'persistently'

File: actor.js [hunk 3]
```diff
@@ -1,44 +1,62 @@
 const Message = require('primea-message')
-const PortManager = require('./portManager.js')
-const DeleteMessage = require('./deleteMessage')
+const CapsManager = require('./capsManager.js')
+const Inbox = require('./inbox.js')
 
-module.exports = class Kernel {
```
Comment (sdtsui) [inline actor.js]: Now that this is called actor instead of kernel, should we update docs/any var names for consistency?

File: actor.js [hunk 4]
```diff
@@ -1,44 +1,62 @@
...
+module.exports = class Actor {
   /**
    * the Kernel manages the varous message passing functions and provides
    * an interface for the containers to use
    * @param {Object} opts
    * @param {Object} opts.id - the UUID of the Kernel
```
Comment (sdtsui) [inline actor.js]: kernel->actor?

File: actor.js [hunk 5]
```diff
@@ -1,44 +1,62 @@
 const Message = require('primea-message')
-const PortManager = require('./portManager.js')
-const DeleteMessage = require('./deleteMessage')
+const CapsManager = require('./capsManager.js')
+const Inbox = require('./inbox.js')
 
-module.exports = class Kernel {
+module.exports = class Actor {
   /**
    * the Kernel manages the varous message passing functions and provides
```
Comment (sdtsui) [inline actor.js]: kernel->actor?

File: actor.js [hunk 6]
```diff
@@ -70,52 +86,47 @@ module.exports = class Kernel {
...
+  /**
+   * Runs the startup routine for the actor
+   */
   startup () {
     return this.container.onStartup()
   }
 
   /**
-   * run the kernels code with a given enviroment
+   * run the Actor with a given message
```
Comment (sdtsui) [inline actor.js]: :+1:

File: actor.js [hunk 7]
```diff
@@ -128,16 +139,16 @@ module.exports = class Kernel {
...
-    const message = new Message(opts)
-    this.ports.checkSendingPorts(message)
-    return message
+  createActor (type, message) {
+    const id = this._generateNextId()
+    return this.hypervisor.createActor(type, message, id)
   }
 
-  generateNextId () {
+  _generateNextId () {
```
Comment (sdtsui) [inline actor.js]: --- sorry, misread the assignment there. Please disregard. makes sense!

File: capsManager.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+module.exports = class CapsManager {
+  /**
+   * The caps manager manages perstantly stores the capabilities
```
Comment (sdtsui) [inline capsManager.js]: 'persistently'

File: capsManager.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+   * Stores a capability persistantly
+   * @param {String} key
+   * @param {Object} cap
+   */
+  store (key, cap) {
+    this._storedCaps[key] = cap
+  }
+
+  /**
+   * gets a cap given it's key
```
Comment (sdtsui) [inline capsManager.js]: its

**Review by sdtsui — COMMENTED** [review]
@wanderer - here is a first pass, as changes come in live.

File: actor.js (hunk 3 above)
Comment (wanderer) [reply actor.js]: yes, defaintly

**Comment by sdtsui** [comment]
adding `.gitignore` w/ https://github.com/primea/js-primea-hypervisor/pull/169

**Pull request merged**